azurerm/resource_arm_automation_account.go

```diff
@@ -169,8 +188,21 @@ func resourceArmAutomationAccountRead(d *schema.ResourceData, meta interface{})
 		d.Set("location", azureRMNormalizeLocation(*location))
 	}
 
-	if err := d.Set("sku", flattenAutomationAccountSku(resp.Sku)); err != nil {
-		return fmt.Errorf("Error setting `sku`: %+v", err)
+	if skuName == "" {
+		// Remove in 2.0
+		if sku := resp.Sku; sku != nil {
```

Comment:
This entire thing can then be:
```
		if sku := resp.Sku; sku != nil {
			if name := sku.Name; name != nil {
				d.Set("sku", string(name)) // todo remove in 2.0
				d.Set("sku_name", string(name))
			}
		}
```

i don't think there is a need for the flatten function anymore

Reply:
I did remove one of the flatten functions, however I don't think we should be setting both of the values.